File: server/dotnet/sdk-dotnet-example/Infrastructure/Extensions/Models/CountryExtension.cs
using Business.Domain.Common.Enums;

namespace Infrastructure.Extensions.Models;

public static class CountryExtension
{
    private static readonly Dictionary<Country, string> CountryIsoCodes = new()
    {
        { Country.England, "GB" },
        { Country.Germany, "DE" },
        { Country.France, "FR" }
    };
    
    public static string? ToIsoAlpha2(this Country? country)
    {
        return country == null ? string.Empty : CountryIsoCodes.GetValueOrDefault(country.Value);
    }
}

File: server/ruby/app/business/extensions/models/country_extension.rb
module Business
  module Extensions
    module Models
      module CountryExtension
        COUNTRY_ISO_CODES = {
          'England' => 'GB',
          'Germany' => 'DE',
          'France'  => 'FR'
        }.freeze

        def self.to_iso_alpha2(country_enum)
          return '' if country_enum.nil?

          COUNTRY_ISO_CODES[country_enum.to_s] || ''
        end
      end
    end
  end
end
